packages/flutter/test/material/data_table_test.dart

```diff
@@ -1629,4 +1629,39 @@ void main() {
       _customHorizontalMargin,
     );
   });
+
+  testWidgets('DataRow does not receive focus by default', (WidgetTester tester) async {
```

Comment:
This really has nothing to do with focus, it has to do with being enabled/disabled.  It really should say something like "DataRow is disabled when `onSelectChanged` is not set" or something similar.